### PR TITLE
Let jekyll build SCSS

### PR DESCRIPTION
**Pull request by gabebw**
According to [this page](http://jekyllrb.com/docs/assets/), Jekyll understands SCSS and will build it automatically (even when running `jekyll serve`).

So:
- I deleted the compiled `styles.css` files everywhere
- Git now ignores `styles.css` files everywhere, and ignores some Jekyll-specific generated files too
- The `styles.scss` is now in the top-level directory so that when it's built, it can be referred to as `/styles.css`

I did not change any CSS styling.

### CODE REVIEW

**Comment by gabebw** [comment]
Actually I'm not 100% sure this works.

File: styles.scss [hunk 1]
```diff
@@ -1,3 +1,6 @@
+---
```
Comment (gabebw) [inline styles.scss]: These dashes are required otherwise Jekyll doesn't convert it from SCSS to CSS.

It's pretty silly.

**Comment by gabebw** [comment]
@foxed It definitely works now.

**Comment by foxed** [comment]
What is the benefit of this? Sorry I haven't done css/scss in so long. I believe you I just want to understand better, nahwutimean??

**Comment by gabebw** [comment]
> What is the benefit of this?

Good question! This PR fixes two things:

1) There are some unused .css files: if you remove them, and change nothing else, the site would still work. Those files are:

* ` _sass/styles.css`
* `styles.css`
* `stylesheets` (which is mostly error messages anyway)

Jekyll is currently reading `public/stylesheets.css` to find the styles, because `/styles.css` in the HTML checks for `_site/styles.css` (which fails) and then  checks for `/public/styles.css` (which works).

Previously, that `public/styles.css` file had to manually generated by the `sass` command-line tool, which was kind of a pain.

So I removed all of those files. But wait! you say. Now where will the styles come from?

2) Jekyll now auto-generates the `styles.scss` file and copies it into `_site` when you run `jekyll build` or `jekyll serve`, so you never have to run `sass` again, just Jekyll. In order for jekyll to compile the SCSS into vanilla CSS, it requires the `---` at the top, otherwise it copies it as-is and doesn't process it at all.

So now Jekyll looks in `_site/styles.css` just like before (which works now) and doesn't check `public/styles.css` at all because it already found the file.

**Comment by foxed** [comment]
Okay, awesome, so it streamlines everything and cleans up some unnecessary overhang. Thank you! @gabebw

**Comment by gabebw** [comment]
Exactly. One file instead of 5, _and_ it automatically builds the CSS with jekyll.

**Pull request closed without merging**